docs/source/caching/cache-key-resolution.mdx

```diff
@@ -8,181 +8,64 @@ The normalized cache computes a **cache key** for each object that is stored in
+Declarative cache key resolution has a few notable quirks and limitations you should be aware of while implementing your `@typePolicy` directives:
 
-## Caveats & limitations
+1. If any type implements two interfaces with conflicting type policies, a validation error will be thrown when executing code generation. This can be corrected by applying a type policy to the type directly to disambiguate the conflict.
 
-Cache key resolution has a few notable quirks and limitations you should be aware of while implementing your cache key resolution function:
+2. When an object is returned for a field of an interface type where the interface has a `@typePolicy`, Apollo iOS will first attempt to find a `@typePolicy` for the concrete type by using the `__typename` field of the returned object. If the type does not have an `@typePolicy` of its own, the interface's `@typePolicy` will be applied.
 
-1. While the cache key for an object can use a field from another nested object, if the fields on the referenced object are changed in another operation, the cache key for the dependent object will not be updated. For nested objects that are not normalized with their own cache key, this will never occur, but if the nested object is an entity with its own cache key, it can be mutated independently. In that case, any other objects whose cache keys are dependent on the mutated entity will not be updated automatically. You must take care to update those entities manually with a cache mutation.
+If the same object is returned for multiple fields of different interface types with conflicting type policies, it is possible that the same object is resolved with two different type policies, leading to cache inconsistencies.
 
-2. The `object` passed to this function represents data for an object in an specific operation model, not a type in your schema. This means that [aliased fields](https://spec.graphql.org/draft/#sec-Field-Alias) will be keyed on their alias name, not the name of the field on the schema type.
+In most circumstances, validation that is run during code generation prevents this from occuring. However this may still occur if changes are made to the schema after code generation runs (or after your application is published).
 
-3. The `object` parameter of this function is an ``ObjectData`` struct that wraps the underlying object data. Because cache key resolution is performed both on raw JSON (from a network response) and `SelectionSet` model data (when writing to the cache directly), the underlying data will have different formats. The ``ObjectData`` wrapper is used to normalize this data to a consistent format in this context.
\ No newline at end of file
+This is possible if:
+  A) A type is added to your schema after code generation is executed and that type implements two different interfaces with different type policies
+  B) Implementation of an interface is added to an existing type which already implements an interface with a different type policy
```

Comment:
```suggestion
In particular, this can happen if:

- A type is added to your schema after code generation execution and that type implements two different interfaces with different type policies.
- Implementation of an interface is added to an existing type that already implements an interface with a different type policy.
```